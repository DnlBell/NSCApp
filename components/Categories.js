import React from 'react';
import { View, TouchableOpacity, Text } from 'react-native';

import styles from '../styles/Catagories'

class Categories extends React.Component{

    constructor() {
        super()
        this.state = {
            catagory : [
                'Academics',
                'Arts & crafts',
                'Camps',
                'Dance & Music',
                'Food & Drink',
                'Health & Well-Being',
                'Home & Gardening',
                'Language, Writing & Humanities',
                'Lifestyle & Leisure',
                'Sports & Activities',
                'Tech & Professional Development',
                'Theater, Film & Television'
            ],
            categoriesView : []
        }

        for(let i = 0; i < this.state.catagory.length; i++){
            this.state.categoriesView.push(
                <View key = {i}>        
                    <TouchableOpacity style = {styles.card}>
                        <Text style ={styles.cardText}>{this.state.catagory[i]}</Text>
                    </TouchableOpacity>
                </View>
            )
        }
    }

    render() {
        return(
            <View>
                <Text style = {styles.text}>Explore Our Categories</Text>
                {this.state.categoriesView}
            </View>
        )
    
    }
}

export default Categories;