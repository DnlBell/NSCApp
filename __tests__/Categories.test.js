import  Categories from '../components/Categories';
import renderer from 'react-test-renderer';
import React from 'react';

it('renders correctly', () => {
  const tree = renderer
    .create(<Categories />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});