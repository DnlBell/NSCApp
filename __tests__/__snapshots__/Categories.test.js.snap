// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#5E5E5E",
        "fontSize": 22,
        "margin": 16,
        "textAlign": "center",
      }
    }
  >
    Explore Our Categories
  </Text>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Academics
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Arts & crafts
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Camps
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Dance & Music
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Food & Drink
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Health & Well-Being
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Home & Gardening
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Language, Writing & Humanities
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Lifestyle & Leisure
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Sports & Activities
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Tech & Professional Development
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5E5E5E",
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 16,
          "opacity": 1,
          "padding": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Theater, Film & Television
      </Text>
    </View>
  </View>
</View>
`;
